src/StoreKeeper/WooCommerce/B2C/Backoffice/MetaBoxes/OrderSyncMetaBox.php

```diff
@@ -229,4 +229,53 @@ protected function getNonceSyncActionLink($post): string
 
         return wp_nonce_url($syncLink, static::ACTION_NAME.'_post_'.$wooCommerceOrder->get_id());
     }
+
+    public function syncAllPaidOrders(): void
+    {
+        if (class_exists('WC_Order_Query')) {
+            $query = new WC_Order_Query([
+                'status' => ['processing', 'completed'],
+                'limit' => -1,
+            ]);
+
+            $orders = $query->get_orders();
+            foreach ($orders as $order) {
+                $orderId = $order->get_id();
+                $noPaidOrder = 'SELECT * FROM wp_storekeeper_pay_orders_payments WHERE order_id = {' . $orderId .'} AND is_paid = 0;';
+
+                if ($noPaidOrder) {
+                    try {
+                        $export = new OrderExport(
+                            [
+                                'id' => $orderId,
+                            ]
+                        );
+
+                        try {
+                            $export->run();
```

Comment:
```suggestion
                        $export = new \StoreKeeper\WooCommerce\B2C\Tools\OrderHandler();
                        $export->create($orderId);
```

Instead of running it directly shedule it. it will run using the task sheduler and than the timeouts/memory leaks are not an issues.
Also errors will be visible in tha tasks tab.